### PR TITLE
Integrate new enigma contract

### PR DESCRIPTION
**Pull request by lenak25**
WIP

### CODE REVIEW

**Comment by lenak25** [comment]
@Isan-Rivkin ready for your review

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "async": "^2.6.1",
     "commander": "^2.18.0",
     "config": "^2.0.1",
+    "dotenv": "^7.0.0",
```
Comment (Isan-Rivkin) [inline package.json]: remove please :)

File: src/ethereum/StateSync.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const DbUtils = require('../common/DbUtils');
  * TODO: as async function returns a Promise, apply both options: return the data using the callback
  * and return the data using the promise
  * */
-async function getRemoteMissingStates(api, localTips, callback) {
+async function getRemoteMissingStates(api, localTips, logger, callback) {
```
Comment (Isan-Rivkin) [inline src/ethereum/StateSync.js]: no logger

**Comment by Isan-Rivkin** [comment]
tAck

**Comment by lenak25** [comment]
Build issues fixed...

**Pull request merged**